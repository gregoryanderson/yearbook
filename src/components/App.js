import React, { Component } from "react";
import Cohort from "./Cohort";
import Form from "./Form";
import peopleData from "../data/yearbook-data.js";
import "./App.css";

class App extends Component {
  constructor() {
    super();
    this.state = {
      staff: peopleData.staff,
      students: peopleData.students
    };
  }

  addNewPerson = newPerson => {
    let addedPerson = { ...newPerson, id: Date.now() };
    this.setState({
      [newPerson.role]: [...this.state[newPerson.role], addedPerson]
    });
  };

  removePerson = person => {
    const filteredStudents = this.state.students.filter(
      human => human.id !== person.id
    );
    const filteredStaff = this.state.staff.filter(
      human => human.id !== person.id
    )
    this.setState({ students: filteredStudents, staff: filteredStaff });
  };

  render = () => {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Turing Yearbook</h1>
        </header>
        <Form addNewPerson={this.addNewPerson} />
        <h1 className="student__h1">Students</h1>
        <Cohort
          people={this.state.staff}
          removePerson={this.removePerson}
          type={"staff"}
        />
        <h1 className="student__h1">Students</h1>
        <Cohort
          people={this.state.students}
          removePerson={this.removePerson}
          type={"students"}
        />
      </div>
    );
  };
}

export default App;
