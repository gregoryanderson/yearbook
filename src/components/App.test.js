import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should update the state when a addNewPerson is called', () => {
    const wrapper = shallow(<App />)
    const newPerson = {
      id: 999, 
      name: 'Greg',
      quote: 'This is pretty cool',
      superlative: 'Most likely to be a raptor',
      photo: "https://placekitten.com/200/300",
      role: "students"
    }
    expect(wrapper.state('students').length).toEqual(26);
    wrapper.instance().addNewPerson(newPerson)
    expect(wrapper.state('students').length).toEqual(27);
  })

  it('should delete a member and update the state when removePerson is called', () => {
    const wrapper = shallow(<App />);
    const newPerson = {
      id: 21, 
      name: 'Abigail', 
      quote: 'This is an original quote', 
      superlative: 'Most likely to be a cat or something', 
      photo: 'https://placekitten.com/200/300'
    }
    expect(wrapper.state('students').length).toEqual(26);
    wrapper.instance().removePerson(newPerson);
    expect(wrapper.state('students').length).toEqual(25);
  })

  it('should match the snapshot taken', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot()
  })
})
