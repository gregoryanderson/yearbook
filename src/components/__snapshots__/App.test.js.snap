// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot taken 1`] = `ShallowWrapper {}`;
